src/pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -591,20 +591,9 @@ def __resolve_ckpt_dir(self, trainer: "pl.Trainer") -> None:
         if trainer._weights_save_path_internal != trainer.default_root_dir:
             # the user has changed weights_save_path
             ckpt_path = os.path.join(trainer._weights_save_path_internal, "checkpoints")
-        elif len(trainer.loggers) > 0:
-            if trainer.loggers[0].save_dir is not None:
-                save_dir = trainer.loggers[0].save_dir
-            else:
-                save_dir = trainer.default_root_dir
-            save_dir = trainer.loggers[0].save_dir or trainer.default_root_dir
-            name = trainer.loggers[0].name
-            version = trainer.loggers[0].version
-            version = version if isinstance(version, str) else f"version_{version}"
-
-            ckpt_path = os.path.join(save_dir, str(name), version, "checkpoints")
         else:
-            # if no loggers, use default_root_dir
-            ckpt_path = os.path.join(trainer.default_root_dir, "checkpoints")
+            log_dir = trainer.log_dir if trainer.log_dir is not None else trainer.default_root_dir
+            ckpt_path = os.path.join(log_dir, "checkpoints")
```

Comment:
We might want to introduce `Trainer.experiment_dir` for this purpose here, then deprecate the old `Trainer.log_dir`. For users who access trainer.log_dir in their code. 

What do you think?

Reply:
Good idea, but I find that this will bring many modifications, should I start another PR or follow this one?

Reply:
Do you mean because trainer.log_dir is used in many places? I think that's fine. If you think it's too much, you could only do the changes in the source code and do the docs updates in a separate PR mabye.

Reply:
I see, will look at it. BTW, it will be much appreciated if you could check the last failed test (`loggers/test_all.py::test_loggers_fit_test_all[CometLogger]`) when you are free, I'm not sure why it failed.

Reply:
I gave it a try. Not 100% sure if my fix is correct, but if I'm looking at it correctly, it's just because the removed logic in model checkpoint caused the test to first access version before experiment, and because we mock a lot of things it didn't like that :)

Reply:
Thx @awaelchli, I'm even less sure because I'm not familiar with the Trainer part yet, so I didn't modify the test directly. 

Also I have submitted the code to introduce `Trainer.experiment_dir`.